judge/admin/contest.py

```diff
@@ -148,7 +159,29 @@ def make_private(self, request, queryset):
 
     def get_urls(self):
         return [url(r'^rate/all/$', self.rate_all_view, name='judge_contest_rate_all'),
-                url(r'^(\d+)/rate/$', self.rate_view, name='judge_contest_rate')] + super(ContestAdmin, self).get_urls()
+                url(r'^(\d+)/rate/$', self.rate_view, name='judge_contest_rate'),
+                url(r'^(\d+)/judge/(\d+)/$', self.rejudge_view, name='judge_contest_rejudge')] + super(ContestAdmin, self).get_urls()
+
+    def rejudge_view(self, request, contest_id, problem_id):
+        if not request.user.has_perm('judge.rejudge_submission'):
+            self.message_user(request, ugettext('You do not have the permission to rejudge submissions.'),
+                              level=messages.ERROR)
+            return
+
+        queryset = ContestSubmission.objects.filter(problem_id=problem_id).select_related('submission')
+        if queryset.count() > getattr(settings, 'REJUDGE_SUBMISSION_LIMIT', 10) and \
+                not request.user.has_perm('judge.rejudge_submission_lot'):
+            self.message_user(request, ugettext('You do not have the permission to rejudge THAT many submissions.'),
+                              level=messages.ERROR)
+            return
+
+        for model in queryset:
+            model.submission.judge(rejudge=True)
+
+        self.message_user(request, ungettext('%d submission were successfully scheduled for rejudging.',
+                                             '%d submissions were successfully scheduled for rejudging.',
+                                             queryset.count()) % queryset.count())
```

Comment:
No you can't use `queryset.count()` here like this without this running 3 queries. Use `len(queryset)`, which reuses the result from the loop above.